include/Types.h

```diff
@@ -18,9 +18,19 @@
 #pragma once
 
 #include <cstdint>
+#include <string.h>
+//#include <exception.h>
+#include <bits/stdc++.h>
 
 namespace Oomd {
 
+class short_lived_exception : public std::exception {
```

Comment:
On second thought, I don't think we need a custom exception type if we're only passing string messages. It might be useful if we needed more context (eg from a vector), but we don't. It's probably cleaner to throw an `std::runtime_error`.